source/backend/api/Services/CompensationRequisitionService.cs

```diff
@@ -73,12 +73,28 @@ public PimsCompensationRequisition Update(PimsCompensationRequisition compensati
 
             CheckDraftStatusUpdateAuthorized(compReqStatusComparable);
             CheckTotalAllowableCompensation(compensationRequisition.AcquisitionFileId, compensationRequisition);
+            compensationRequisition.FinalizedDate = CheckFinalizedDate(compReqStatusComparable, currentCompensation.FinalizedDate);
 
             PimsCompensationRequisition updatedEntity = _compensationRequisitionRepository.Update(compensationRequisition);
             AddNoteIfStatusChanged(compensationRequisition.Internal_Id, compensationRequisition.AcquisitionFileId, compReqStatusComparable);
             _compensationRequisitionRepository.CommitTransaction();
 
             return updatedEntity;
+
+            DateTime? CheckFinalizedDate((bool? currentStatusIsDraft, bool? newStatusIsDraft) statusComparable, DateTime? currentValue)
+            {
+                if (statusComparable.currentStatusIsDraft.Equals(statusComparable.newStatusIsDraft))
+                {
+                    return currentValue;
+                }
+
+                if(statusComparable.newStatusIsDraft.HasValue)
+                {
+                    return statusComparable.newStatusIsDraft.Value ? null : DateTime.UtcNow;
+                }
+
+                return null;
+            }
```

Comment:
question: is this a function within another function body? if so, I don't think we follow that pattern.

Reply:
yes, it's a local function. I like it because it's only useful to the method that contains it.